### PR TITLE
rebuild schema's on error

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?229467
@czue 

This will rebuild the schema but also silently exclude it from the bulk export. There isn't really any mechanism form giving feedback to the user and adding it wouldn't be simple. 

Considering new exports should be out soon don't think it's worth the time.

### CODE REVIEW

**Comment by snopoke** [comment]
I guess one option would be to fail the entire export and force the user to try it again.

**Comment by czue** [comment]
this seems fine since new exports are coming out very soon. another thing you could do is write an error message to the export file (up to you whether that seems worth it)

**Comment by snopoke** [comment]
@czue updated

**Comment by czue** [comment]
:+1: on :white_check_mark:

File: corehq/form_processor/tests/test_case_dbaccessor.py [hunk 1]
```diff
@@ -102,43 +102,17 @@ def test_get_indices(self):
...
-                relationship_id=CommCareCaseIndexSQL.CHILD
-            )
-            case.track_create(index1)
-
-            CaseAccessorSQL.save_case(case)
-            return case.case_id
-
         referenced_case_ids = [uuid.uuid4().hex, uuid.uuid4().hex]
-        expected_case_ids = [_create_case_with_index(case_id) for case_id in referenced_case_ids]
+        _create_case_with_index(referenced_case_ids[0], case_is_deleted=True)  # case shouldn't be included in results
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_case_dbaccessor.py]: E501 line too long (118 > 115 characters)

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**